virtcontainers/qemu.go

```diff
@@ -169,6 +174,7 @@ func (q *qemu) init(sandbox *Sandbox) error {
 		return err
 	}
 
+	q.vmConfig = sandbox.config.VMConfig
```

Comment:
The memory coming through the `Resources` structure in `sandbox.config.VMConfig` is part of some legacy code that should be removed IMO. We need a unified way to provide the resources, and that's using `HypervisorConfig` that we already have through `q.config`.
It is a bit odd to have CPU and Memory resources provided through 2 different structures.

Reply:
Why don't you do that directly as part of this PR (separate commit of course). Do you think that's a lot of work ?

Reply:
As discussed in the arch meeting, it will break ondisk metadata compatibility so I prefer to defer it to a standalone PR at least.

Reply:
As discussed during Arch committee meeting, this rework will be done in a separate PR to prevent from API breakage.